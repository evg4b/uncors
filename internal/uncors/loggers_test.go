package uncors_test

import (
	"bytes"
	"testing"

	"github.com/charmbracelet/log"
	"github.com/evg4b/uncors/internal/tui/styles"
	"github.com/evg4b/uncors/internal/uncors"
	"github.com/evg4b/uncors/testing/testutils"
	"github.com/stretchr/testify/assert"
)

func TestLoggers(t *testing.T) {
	t.Run("NewProxyLogger", testutils.LogTest(func(t *testing.T, output *bytes.Buffer) {
		log.SetLevel(log.DebugLevel)

		tests := []struct {
			name     string
			logger   *log.Logger
			expected string
		}{
			{
				name:     "ProxyLogger",
				logger:   uncors.NewProxyLogger(log.Default()),
				expected: styles.ProxyStyle.Render("PROXY"),
			},
			{
				name:     "MockLogger",
				logger:   uncors.NewMockLogger(log.Default()),
				expected: styles.MockStyle.Render("MOCK"),
			},
			{
				name:     "StaticLogger",
				logger:   uncors.NewStaticLogger(log.Default()),
				expected: styles.StaticStyle.Render("STATIC"),
			},
			{
				name:     "CacheLogger",
				logger:   uncors.NewCacheLogger(log.Default()),
				expected: styles.CacheStyle.Render("CACHE"),
			},
		}
		for _, testCase := range tests {
			t.Run(testCase.name, func(t *testing.T) {
				t.Run("Error", testutils.UniqOutput(output, func(t *testing.T, output *bytes.Buffer) {
					testCase.logger.Error("error")

					assert.Contains(t, output.String(), testCase.expected)
				}))

				t.Run("Warn", testutils.UniqOutput(output, func(t *testing.T, output *bytes.Buffer) {
					testCase.logger.Warn("warn")

					assert.Contains(t, output.String(), testCase.expected)
				}))

				t.Run("Info", testutils.UniqOutput(output, func(t *testing.T, output *bytes.Buffer) {
					testCase.logger.Info("info")

					assert.Contains(t, output.String(), testCase.expected)
				}))

				t.Run("Debug", testutils.UniqOutput(output, func(t *testing.T, output *bytes.Buffer) {
					testCase.logger.Debug("debug")

					assert.Contains(t, output.String(), testCase.expected)
				}))

				t.Run("Print", testutils.UniqOutput(output, func(t *testing.T, output *bytes.Buffer) {
					testCase.logger.Print("print")

					assert.Contains(t, output.String(), testCase.expected)
				}))
			})
		}
	}))
}
