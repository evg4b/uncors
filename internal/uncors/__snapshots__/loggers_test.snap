
[TestLoggers/NewProxyLogger/ProxyLogger/Error - 1]
[1m[48;2;105;113;247m [0m[38;2;0;0;0;48;2;105;113;247mPROXY[0m[48;2;105;113;247m [0m[48;2;105;113;247m [0m  ERROR  [0m error

---

[TestLoggers/NewProxyLogger/ProxyLogger/Warn - 1]
[1m[48;2;105;113;247m [0m[38;2;0;0;0;48;2;105;113;247mPROXY[0m[48;2;105;113;247m [0m[48;2;105;113;247m [0m  WARN   [0m warn

---

[TestLoggers/NewProxyLogger/ProxyLogger/Info - 1]
[1m[48;2;105;113;247m [0m[38;2;0;0;0;48;2;105;113;247mPROXY[0m[48;2;105;113;247m [0m[48;2;105;113;247m [0m  INFO   [0m info

---

[TestLoggers/NewProxyLogger/ProxyLogger/Debug - 1]
[1m[48;2;105;113;247m [0m[38;2;0;0;0;48;2;105;113;247mPROXY[0m[48;2;105;113;247m [0m[48;2;105;113;247m [0m  DEBUG  [0m debug

---

[TestLoggers/NewProxyLogger/ProxyLogger/Print - 1]
[48;2;105;113;247m [0m[38;2;0;0;0;48;2;105;113;247mPROXY[0m[48;2;105;113;247m [0m[48;2;105;113;247m [0m print

---

[TestLoggers/NewProxyLogger/MockLogger/Error - 1]
[1m[48;2;238;127;248m [0m[38;2;0;0;0;48;2;238;127;248mMOCK[0m[48;2;238;127;248m [0m[48;2;238;127;248m  [0m  ERROR  [0m error

---

[TestLoggers/NewProxyLogger/MockLogger/Warn - 1]
[1m[48;2;238;127;248m [0m[38;2;0;0;0;48;2;238;127;248mMOCK[0m[48;2;238;127;248m [0m[48;2;238;127;248m  [0m  WARN   [0m warn

---

[TestLoggers/NewProxyLogger/MockLogger/Info - 1]
[1m[48;2;238;127;248m [0m[38;2;0;0;0;48;2;238;127;248mMOCK[0m[48;2;238;127;248m [0m[48;2;238;127;248m  [0m  INFO   [0m info

---

[TestLoggers/NewProxyLogger/MockLogger/Debug - 1]
[1m[48;2;238;127;248m [0m[38;2;0;0;0;48;2;238;127;248mMOCK[0m[48;2;238;127;248m [0m[48;2;238;127;248m  [0m  DEBUG  [0m debug

---

[TestLoggers/NewProxyLogger/MockLogger/Print - 1]
[48;2;238;127;248m [0m[38;2;0;0;0;48;2;238;127;248mMOCK[0m[48;2;238;127;248m [0m[48;2;238;127;248m  [0m print

---

[TestLoggers/NewProxyLogger/StaticLogger/Error - 1]
[1m[48;2;166;252;157m [0m[38;2;0;0;0;48;2;166;252;157mSTATIC[0m[48;2;166;252;157m [0m  ERROR  [0m error

---

[TestLoggers/NewProxyLogger/StaticLogger/Warn - 1]
[1m[48;2;166;252;157m [0m[38;2;0;0;0;48;2;166;252;157mSTATIC[0m[48;2;166;252;157m [0m  WARN   [0m warn

---

[TestLoggers/NewProxyLogger/StaticLogger/Info - 1]
[1m[48;2;166;252;157m [0m[38;2;0;0;0;48;2;166;252;157mSTATIC[0m[48;2;166;252;157m [0m  INFO   [0m info

---

[TestLoggers/NewProxyLogger/StaticLogger/Debug - 1]
[1m[48;2;166;252;157m [0m[38;2;0;0;0;48;2;166;252;157mSTATIC[0m[48;2;166;252;157m [0m  DEBUG  [0m debug

---

[TestLoggers/NewProxyLogger/StaticLogger/Print - 1]
[48;2;166;252;157m [0m[38;2;0;0;0;48;2;166;252;157mSTATIC[0m[48;2;166;252;157m [0m print

---

[TestLoggers/NewProxyLogger/CacheLogger/Error - 1]
[1m[48;2;254;252;127m [0m[38;2;0;0;0;48;2;254;252;127mCACHE[0m[48;2;254;252;127m [0m[48;2;254;252;127m [0m  ERROR  [0m error

---

[TestLoggers/NewProxyLogger/CacheLogger/Warn - 1]
[1m[48;2;254;252;127m [0m[38;2;0;0;0;48;2;254;252;127mCACHE[0m[48;2;254;252;127m [0m[48;2;254;252;127m [0m  WARN   [0m warn

---

[TestLoggers/NewProxyLogger/CacheLogger/Info - 1]
[1m[48;2;254;252;127m [0m[38;2;0;0;0;48;2;254;252;127mCACHE[0m[48;2;254;252;127m [0m[48;2;254;252;127m [0m  INFO   [0m info

---

[TestLoggers/NewProxyLogger/CacheLogger/Debug - 1]
[1m[48;2;254;252;127m [0m[38;2;0;0;0;48;2;254;252;127mCACHE[0m[48;2;254;252;127m [0m[48;2;254;252;127m [0m  DEBUG  [0m debug

---

[TestLoggers/NewProxyLogger/CacheLogger/Print - 1]
[48;2;254;252;127m [0m[38;2;0;0;0;48;2;254;252;127mCACHE[0m[48;2;254;252;127m [0m[48;2;254;252;127m [0m print

---
